### PR TITLE
Mbedtls kconfig cleanup

### PR DESCRIPTION
**Pull request by tejlmand**
This is a series of Kconfig cleanups related to mbedTLS to remove occurrences of stuck symbol syndrome by allowing other Kconfig files to disable the mbedTLS prompt when mbedTLS is automatically selected, in the cases where a config change should also ensure a de-selection of mbed TLS.

### CODE REVIEW

File: subsys/net/l2/openthread/Kconfig [hunk 1]
```diff
@@ -149,9 +155,32 @@ config OPENTHREAD_SHELL
...
+
+choice OPENTHREAD_SECURITY
+	prompt "OpenThread security"
+	default OPENTHREAD_MBEDTLS_CHOICE
+	depends on !OPENTHREAD_SECURITY_INTERNAL
+
+config CUSTOM_OPENTHREAD_SECURITY
+	bool "Custom"
+	help
+	  Security settings will be controlled directly by this user.
```
Comment (edmont) [inline subsys/net/l2/openthread/Kconfig:170]:
```suggestion
	  Security settings will be controlled directly by the user.
```

File: subsys/net/l2/openthread/Kconfig [hunk 2]
```diff
@@ -149,9 +155,32 @@ config OPENTHREAD_SHELL
...
+	  Enabling this setting will give access to full control of mbedTLS
+	  configuration.
+
+config OPENTHREAD_MBEDTLS_CHOICE
+	bool "bedTLS built-in"
```
Comment (edmont) [inline subsys/net/l2/openthread/Kconfig:175]:
```suggestion
	bool "mbedTLS built-in"
```

**Review by edmont — APPROVED** [review]
Thanks!

File: modules/Kconfig.mbedtls [hunk 3]
```diff
@@ -3,8 +3,17 @@
...
+	help
+	  Symbol to disable the prompt for MBEDTLS selection.
+	  This symbol may be used internally in a Kconfig tree to hide the
+	  mbedTLS menu prompt and instead handle the selection of MBEDTLS from
+	  dependent sub-configurations and thus preven stuck symbol behavior.
+
+
 menuconfig MBEDTLS
-	bool "mbedTLS Support"
+	bool "mbedTLS Support" if !MBEDTLS_PROMPTLESS
```
Comment (rlubos) [inline modules/Kconfig.mbedtls:16]: I've tried this with `echo_server`/`echo_client` samples and I can still see this entry even though `MBEDTLS_PROMPTLESS` is set to `y` (the default configuration). What am I missing?

Comment (tejlmand) [reply modules/Kconfig.mbedtls:16]: The fact that `mbedTLS` is a menuconfig and not a config.
Thus, when another Kconfig selects the promptless menuconfig, you will see the symbol in order to access the rest of the menu.

But if `MBEDTLS=n` and  `MBEDTLS_PROMPTLESS=y` then you will not see the menu and cannot enable it.

If `MBEDTLS_PROMPTLESS=n` you can of course both see, configure and enter the menu.

**Review by rlubos — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

**Review by ceolin — APPROVED** [review]
Just remember to update the revision before merging it.

File: modules/Kconfig.mbedtls [hunk 4]
```diff
@@ -29,8 +38,15 @@ config MBEDTLS_LIBRARY
 
 endchoice
 
+config CUSTOM_MBEDTLS_CFG_FILE
+	bool "Custom mbed TLS configuration file"
+	help
+	  Allow user defined input for the MBEDTLS_CFG_FILE setting.
+	  You can specify the actual configuration file using in
```
Comment (pfalcon) [inline modules/Kconfig.mbedtls:45]: >  You can specify the actual configuration file using in
	  MBEDTLS_CFG_FILE setting.

Apparently, a typo. Should be "You can specify the actual configuration file using **the** MBEDTLS_CFG_FILE setting."

Comment (tejlmand) [reply modules/Kconfig.mbedtls]: Thanks for spotting. Fixed.

**Pull request merged**